tests/pytests/test_numbers.py

```diff
@@ -389,3 +389,356 @@ def testNegativeValues(env):
+    res2 = env.cmd('FT.SEARCH', 'idx', '@n:[-inf (-3.14] | @n:[(-3.14 +inf]',
+                   'NOCONTENT', 'LIMIT', 0, 20, 'SORTBY', 'n', 'ASC')
+    env.assertEqual(res2, res1)
+    res2 = env.cmd('FT.SEARCH', 'idx', '@n:!=+$n', 'NOCONTENT',
+                   'PARAMS', 2, 'n', '-3.14', 'LIMIT', 0, 20,
+                   'SORTBY', 'n', 'ASC')
+    env.assertEqual(res2, res1)
+
+    # Test != +inf
+    res1 = env.cmd('FT.SEARCH', 'idx', '@n:!=inf', 'NOCONTENT',
+                   'SORTBY', 'n', 'ASC')
+    env.assertEqual(res1, [10, 'key12', 'key7', 'key6', 'key9', 'key8', 'key1',
+                           'key2', 'key3', 'key4', 'key5'])
+    res2 = env.cmd('FT.SEARCH', 'idx', '@n:!=+inf', 'NOCONTENT',
+                   'SORTBY', 'n', 'ASC')
+    env.assertEqual(res2, res1)
+    res2 = env.cmd('FT.SEARCH', 'idx', '@n:!=$n', 'NOCONTENT', 'PARAMS', 2,
+                   'n', 'inf', 'SORTBY', 'n', 'ASC')
+    env.assertEqual(res2, res1)
+    res2 = env.cmd('FT.SEARCH', 'idx', '@n:!=$n', 'NOCONTENT', 'PARAMS', 2,
+                   'n', '+inf', 'SORTBY', 'n', 'ASC')
+    env.assertEqual(res2, res1)
+    res2 = env.cmd('FT.SEARCH', 'idx', '@n:!=+$n', 'NOCONTENT', 'PARAMS', 2,
+                     'n', 'inf', 'SORTBY', 'n', 'ASC')
+    env.assertEqual(res2, res1)
+
+    # Test != -inf
+    res1 = env.cmd('FT.SEARCH', 'idx', '@n:!=-inf', 'NOCONTENT', 'LIMIT', 0, 20,
+                   'SORTBY', 'n', 'ASC')
+    env.assertEqual(res1, [11, 'key7', 'key6', 'key9', 'key8', 'key1', 'key2',
+                           'key3', 'key4', 'key5', 'key10', 'key11'])
+    res2 = env.cmd('FT.SEARCH', 'idx', '@n:!=$n', 'NOCONTENT',
+                   'PARAMS', 2, 'n', '-inf', 'LIMIT', 0, 20,
+                   'SORTBY', 'n', 'ASC')
+    env.assertEqual(res2, res1)
+
+    # Test range and operator in the same query
+    res1 = env.cmd('FT.SEARCH', 'idx', '@n:==3.14 | @n:[10 13]', 'NOCONTENT')
+    env.assertEqual(res1, [4, 'key1', 'key2', 'key3', 'key8'])
+    res2 = env.cmd('FT.SEARCH', 'idx', '@n:[3.14 3.14] | @n:>=10 @n:<=13', 'NOCONTENT')
+    env.assertEqual(res2, res1)
+
+    res1 = env.cmd('FT.SEARCH', 'idx', '@n:==11 | @n:[10 13]', 'NOCONTENT')
+    env.assertEqual(res1, [3, 'key1', 'key2', 'key3'])
+    res2 = env.cmd('FT.SEARCH', 'idx', '@n:[11 11] | @n:>=10 @n:<=13', 'NOCONTENT')
+    env.assertEqual(res2, res1)
+
+    # Test contradiction query
+    res1 = env.cmd('FT.SEARCH', 'idx', '@n:==3.14 @n:!=3.14', 'NOCONTENT')
+    env.assertEqual(res1, [0])
+
+    res1 = env.cmd('FT.SEARCH', 'idx', '@n:==12 @n:==11', 'NOCONTENT')
+    env.assertEqual(res1, [0])
+
+    res1 = env.cmd('FT.SEARCH', 'idx', '@n:<10 @n:>12', 'NOCONTENT')
+    env.assertEqual(res1, [0])
+
+    res1 = env.cmd('FT.SEARCH', 'idx', '@n:==10 @n:[12 15]', 'NOCONTENT')
+    env.assertEqual(res1, [0])
+
+    # Invalid syntax
+    for operator in ['==', '!=', '>', '>=', '<', '<=']:
+        env.expect('FT.SEARCH', 'idx', '@n:' + operator + '(105').error()
```

Comment:
Intended to use a `+`?

Reply:
Yes. Fixed in [12a0967](https://github.com/RediSearch/RediSearch/pull/4516/commits/12a096748e6657d5e892317ea49b6322a8fb8cd9)